lib/standard/text/abstract_text.nit

```diff
@@ -231,14 +231,99 @@ abstract class Text
 	#     assert "abcd".has_suffix("bcd")	     ==  true
 	fun has_suffix(suffix: String): Bool do return has_substring(suffix, length - suffix.length)
 
-	# If `self` contains only digits, return the corresponding integer
+	# Removes the underscores in `self`
+	#
+	#     assert "__init__".remove_underscores == "init"
+	fun remove_underscores: Text do
+		var b = new Buffer
+		for i in chars do
+			if i == '_' then continue
+			b.add i
+		end
+		return b.to_s
+	end
+
+	# Is `self` a well-formed Integer (i.e. parsable via `to_i`)
+	#
+	#     assert "123".is_int
+	#     assert "0b1011".is_int
+	#     assert not "0x_".is_int
+	#     assert not "0xGE".is_int
+	fun is_int: Bool do
+		var s = remove_underscores
+		var pos = 0
+		while s[pos] == '-' do
+			pos += 1
+		end
+		s = s.substring_from(pos)
+		var rets = s.strip_numhead
+		if rets == "" then return false
+		var hd = get_numhead
+		if hd == "0x" or hd == "0X" then return rets.is_hex
+		if hd == "0b" or hd == "0B" then return rets.is_bin
+		if hd == "0o" or hd == "0O" then return rets.is_oct
+		return hd.is_dec
+	end
+
+	# Removes the numeric head of `self` if present
+	fun strip_numhead: Text do
+		if get_numhead != "" then return substring_from(2)
+		return self
+	end
+
+	# Gets the numeric head of `self` if present
+	# Returns "" otherwise
+	#
+	#     assert "0xFEFF".get_numhead  == "0x"
+	#     assert "0b01001".get_numhead == "0b"
+	#     assert "0o872".get_numhead   == "0o"
+	#     assert "98".get_numhead      == ""
+	fun get_numhead: Text do
```

Comment:
Both `strip_numhead` and `get_numhead` should be private, they are very specialized and much less friendly than `is_bin, is_dec...`.